lib/config/dynamic/admiral/source.go

```diff
@@ -352,22 +355,18 @@ func (a *source) whitelist(ctx context.Context, c *client.Admiral, hostProjs []s
 		return nil, nil
 	}
 
-	regs, err := c.ConfigRegistries.GetConfigRegistries(config_registries.NewGetConfigRegistriesParamsWithContext(ctx).WithExpand(&trueStr))
+	regs, err := c.ConfigRegistries.GetConfigRegistriesID(config_registries.NewGetConfigRegistriesIDParamsWithContext(ctx).WithID(defaultVicRegistry))
 	if err != nil {
 		return nil, err
 	}
-
-	var wl []string
-	for _, r := range regs.Payload.Documents {
-		m := &models.ComVmwareAdmiralServiceCommonRegistryServiceRegistryState{}
-		if err := mapstructure.Decode(r, m); err != nil {
-			log.Warnf("skipping registry: %s", err)
-			continue
-		}
-
-		wl = append(wl, m.Address)
+	if regs.Payload == nil {
+		// No default VIC registry configured.
+		return nil, nil
```

Comment:
can we update the calling method to record at Warn level that no default registry was found despite content trust being configured.

Reply:
There's another place in this func where we return `nil, nil` when there are no Admiral projects with a VCH and content trust enabled (https://github.com/vmware/vic/pull/6919/files#diff-1a00736b9f8c89eb8f1862400f7c8a0fR355) - so I've added a warn-level log to this function itself. The caller logs the received whitelist at debug level.